cli/admin/src/test/java/io/pravega/cli/admin/controller/AbstractControllerMetadataCommandsTest.java

```diff
@@ -343,6 +354,55 @@ public void testControllerMetadataViewReaderInfoCommandWithException() throws Ex
         Assert.assertTrue(commandResult.contains("Exception accessing to reader metadata"));
     }
 
+    private CreateReaderGroupEvent buildCreateRGEvent(String scope, String rgName, ReaderGroupConfig config,
+                                                      final long requestId, final long createTimestamp) {
+        Map<String, RGStreamCutRecord> startStreamCuts = config.getStartingStreamCuts().entrySet().stream()
+                                                               .collect(Collectors.toMap(e -> e.getKey().getScopedName(),
+                                                                       e -> new RGStreamCutRecord(ImmutableMap.copyOf(ModelHelper.getStreamCutMap(e.getValue())))));
+        Map<String, RGStreamCutRecord> endStreamCuts = config.getEndingStreamCuts().entrySet().stream()
+                                                             .collect(Collectors.toMap(e -> e.getKey().getScopedName(),
+                                                                     e -> new RGStreamCutRecord(ImmutableMap.copyOf(ModelHelper.getStreamCutMap(e.getValue())))));
+        return new CreateReaderGroupEvent(requestId, scope, rgName, config.getGroupRefreshTimeMillis(),
+                config.getAutomaticCheckpointIntervalMillis(), config.getMaxOutstandingCheckpointRequest(),
+                config.getRetentionType().ordinal(), config.getGeneration(), config.getReaderGroupId(),
+                startStreamCuts, endStreamCuts, createTimestamp);
+    }
+
+    @Test
+    public void testControllerMetadataViewPendingEventsCommand() throws Exception {
+        final String host = UUID.randomUUID().toString();
+        final String requestId = UUID.randomUUID().toString();
+
+        @Cleanup("shutdownNow")
+        ScheduledExecutorService executorService = ExecutorServiceHelpers.newScheduledThreadPool(1, "cli-test");
+        ZKHelper zkHelper = ZKHelper.create(SETUP_UTILS.getZkTestServer().getConnectString(), "pravega-cluster");
+        ZKHostIndex hostIndex = zkHelper.getZkHostIndex(executorService);
+        EventHelper eventHelper = new EventHelper(executorService, host, hostIndex);
+
+        String scopedStreamName = "scope/stream";
+        ReaderGroupConfig rgConf = ReaderGroupConfig.builder().disableAutomaticCheckpoints()
+                                                    .stream(scopedStreamName)
+                                                    .retentionType(ReaderGroupConfig.StreamDataRetention.NONE)
+                                                    .build();
+        CreateReaderGroupEvent createEventOne = buildCreateRGEvent("scope", "rg1", rgConf, 1L, System.currentTimeMillis());
+
+        eventHelper.addRequestToIndex(host, requestId, createEventOne);
```

Comment:
This returns a `CompletableFuture`. So, it may happen that the subsequent lines of test assume this to be executed to be successful, but it may not happen as you are not waiting for the future to complete. This can make the test flaky.

Reply:
Correct Raul,  I have introduced join() for all completableFuture calls it iwll wait for it to complete before asserting.